src/swap/saga.ts

```diff
@@ -247,8 +250,182 @@ export function* swapSubmitSaga(action: PayloadAction<SwapInfo>) {
+  })
+
+  const swapApproveContext = newTransactionContext(TAG, 'Swap/Approve')
+  const swapExecuteContext = newTransactionContext(TAG, 'Swap/Execute')
+
+  const defaultSwapExecuteProps = {
+    toToken: buyTokenAddress,
+    fromToken: sellTokenAddress,
+    amount,
+    amountType,
+    price,
+    allowanceTarget,
+    estimatedPriceImpact,
+    provider: action.payload.quote.rawSwapResponse.details.swapProvider,
+    fromTokenBalance,
+    swapExecuteTxId: swapExecuteContext.id,
+    swapApproveTxId: swapApproveContext.id,
+    estimatedSellTokenUsdValue,
+    estimatedBuyTokenUsdValue,
+  }
+
+  let quoteToTransactionElapsedTimeInMs: number | undefined
+
+  const getTimeMetrics = (): SwapTimeMetrics => ({
+    quoteToUserConfirmsSwapElapsedTimeInMs: swapSubmittedAt - quoteReceivedAt,
+    quoteToTransactionElapsedTimeInMs,
+  })
+
+  try {
+    // Navigate to swap pending screen
+    navigate(Screens.SwapExecuteScreen)
+
+    const wallet = yield* call(getViemWallet, networkConfig.viemChain.celo)
+    if (!wallet.account) {
+      // this should never happen
+      throw new Error('no account found in the wallet')
+    }
+
+    const preparedTransactions = action.payload.quote.preparedTransactions
+    if (preparedTransactions?.type !== 'possible') {
+      // Should never happen
+      throw new Error('No prepared transactions possible')
+    }
+
+    // @ts-ignore typed-redux-saga erases the parameterized types causing error, we can address this separately
+    let nonce: number = yield* call(getTransactionCount, wallet, {
+      address: wallet.account.address,
+      blockTag: 'pending',
+    })
+
+    // Unlock account before executing tx
+    yield* call(unlockAccount, wallet.account.address)
+
+    // Execute transaction(s)
+    yield* put(swapExecute())
+    Logger.debug(TAG, `Starting to swap execute for address: ${wallet.account.address}`)
+
+    const beforeSwapExecutionTimestamp = Date.now()
+    quoteToTransactionElapsedTimeInMs = beforeSwapExecutionTimestamp - quoteReceivedAt
+
+    const txHashes: Hash[] = []
+    for (const preparedTransaction of preparedTransactions.transactions) {
+      const signedTx = yield* call([wallet, 'signTransaction'], {
+        ...preparedTransaction,
+        nonce: nonce++,
+      } as any)
+      const hash = yield* call([wallet, 'sendRawTransaction'], {
+        serializedTransaction: signedTx,
+      })
+      txHashes.push(hash)
+    }
+
+    Logger.debug(TAG, 'Successfully sent swap transaction(s) to the network', txHashes)
+
+    const swapTxHash = txHashes[txHashes.length - 1]
+
+    yield* put(
+      addStandbyTransaction({
+        context: swapExecuteContext,
+        __typename: 'TokenExchangeV3',
+        networkId: networkConfig.defaultNetworkId,
+        type: TokenTransactionTypeV2.SwapTransaction,
+        inAmount: {
+          value: valueToBigNumber(sellAmount)
+            .multipliedBy(guaranteedPrice)
+            .shiftedBy(-toToken.decimals),
```

Comment:
I'm confused.. should this be "fromToken"? Seems like you put "fromToken" in and get "toToken" out for a swap, right? But this does the opposite..

Reply:
Yes I was confused at first too. I double checked and `inAmount` is what the user account received, which is the swap `toToken`.

Reply:
the in and out is relative to the wallet here 😅 it is a bit confusing, but this seems right. the net incoming amount to the wallet is the sellAmount (fromToken) * price (this logic assumes that we are using the sell flow, where we've specified the sell token amount only)